pkg/dao/repository_configs.go

```diff
@@ -820,6 +821,200 @@ func (r repositoryConfigDaoImpl) bulkDelete(ctx context.Context, tx *gorm.DB, or
 	}
 }
 
+func (r repositoryConfigDaoImpl) BulkExport(ctx context.Context, orgID string, reposToExport api.RepositoryExportRequest) ([]api.RepositoryExportResponse, error) {
+	var repoConfigs []models.RepositoryConfiguration
+
+	uuidsValid, uuid := checkForValidRepoUuids(ctx, reposToExport.RepositoryUuids, r.db)
+	if !uuidsValid {
+		return []api.RepositoryExportResponse{}, &ce.DaoError{
+			NotFound: true,
+			Message:  "Could not find repository with UUID: " + uuid,
+		}
```

Comment:
yes this will reject the request as soon as it sees one invalid uuid. i think we'd want to stop checking as soon as one is invalid though and respond with a 400, right?

Reply:
we could instead use just one query to check if all uuids are valid, which would be better!

Reply:
Yes, that is what I meant. Because if someone sends a request to export `2n` repositories, while the first `n` would exist, but the other `n` would not...that means that before successfully exporting the repositories, they would have to send `n` requests for exporting to find out which uuids exist.
I am okay with you leaving it as is. As I said, this will probably be fine on the IB Blueprints side. In our use case, I do not think it would happen that the repository is not found unless some unexpected deletion happens a few milliseconds before exporting them.